### PR TITLE
Expose model persistence refresh icon when model is still in creating state

### PR DESCRIPTION
**Pull request by noahmoss**
Little tweak to expose the model persistence "refresh" icon even before the model has been initially persisted, so that you can manually trigger it and not wait for the refresh task to run. 

Slack discussion: https://metaboat.slack.com/archives/C01LQQ2UW03/p1729542134085539

Before:

After:

I've also added a "disabled" state for users who aren't admins or DB managers.

### CODE REVIEW

**Comment by noahmoss** [comment]
* **#48961** <a href="https://app.graphite.dev/github/pr/metabase/metabase/48961?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @noahmoss and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx [hunk 1]
```diff
@@ -59,6 +59,11 @@ export function ModelCacheManagementSection({ model }: Props) {
   const isError = persistedModel?.state === "error";
   const lastRefreshTime = moment(persistedModel?.refresh_end).fromNow();
 
+  const canRefreshCache =
+    persistedModel && checkCanRefreshModelCache(persistedModel);
+  const refreshTooltip =
+    persistedModel?.state === "creating" ? t`Create now` : t`Refresh now`;
```
Comment (rafpaf) [inline frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx:65]: *Thinking out loud. This isn't a suggestion.*

If you move this ternary to line 91, in place of `refreshTooltip`, you can avoid using `?.` which might simplify the code in one respect. But that means moving the ternary into the JSX, which complicates the code in another respect.

**Comment by noahmoss** [comment]
@rafpaf Switched it to just saying `Create now` which I think feels reasonably clear in the context

File: frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx [hunk 2]
```diff
@@ -59,6 +61,38 @@ export function ModelCacheManagementSection({ model }: Props) {
...
+  const refreshButton =
+    persistedModel?.state === "creating" ? (
+      <CreateOrRefreshButton
+        onClick={() => onRefresh(model.id())}
+      >{t`Create now`}</CreateOrRefreshButton>
+    ) : (
+      <CreateOrRefreshButton onClick={() => onRefresh(model.id())}>
+        <RefreshIcon name="refresh" tooltip={t`Refresh now`} />
+      </CreateOrRefreshButton>
+    );
```
Comment (rafpaf) [inline frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx:76]: I don't think there's anything wrong with this approach but you could instead define a little component:

```
const RefreshButton = ({isModelBeingCreated}: {isModelBeingCreated: boolean}) => 
	<CreateOrRefreshButton onClick={() => onRefresh(model.id())}>
	  {isModelBeingCreated ? t`Create now` : <RefreshIcon name="refresh" tooltip={t`Refresh now`} />}
	</CreateOrRefreshButton>
```

**Review by rafpaf — APPROVED** [review]
Looks terrific!

File: frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx (hunk 2 above)
Comment (rafpaf) [reply frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx:76]: You could also call this outer component `CreateOrRefreshButton` and the styled component `StyledCreateOrRefreshButton`. Or for simplicity delete the styled component and use a `Button`:

```ts
import { Button } from "metabase/ui";

const CreateOrRefreshButton = ({isModelBeingCreated}: {isModelBeingCreated: boolean}) => 
	<Button onClick={() => onRefresh(model.id())} c="var(--mb-color-text-medium)">
	  {isModelBeingCreated ? t`Create now` : <RefreshIcon name="refresh" tooltip={t`Refresh now`} />}
	</Button>
```

The `Button` component has a `variant` prop you could experiment with - maybe try setting it to "subtle". I copied over the color from the styled component but you probably don't need to use that color.

Comment (iethree) [reply frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx:61]: I did this and deleted all the other styled components while I was at it

https://github.com/metabase/metabase/pull/48961/commits/336b717d6825d4a5c3f575d859f4711dbc7df9fc

**Review by iethree — APPROVED** [review]

**Pull request merged**